File: ios/BestChanceStudio/BestChanceStudio/Models/GradeCalculator.swift
import Foundation

struct GradeCalculator {
    static let gradeOrder = ["A+", "A", "B", "C", "D"]

    static func grade(for totalScore: Int, using thresholds: [String: GradeThreshold]) -> (grade: String, label: String) {
        for g in gradeOrder {
            if let t = thresholds[g], totalScore >= t.min, totalScore <= t.max {
                return (g, t.label)
            }
        }
        return ("D", thresholds["D"]?.label ?? "Needs significant improvement.")
    }

    static func gradeColor(for grade: String) -> GradeColorInfo {
        switch grade {
        case "A+", "A":
            return GradeColorInfo(foreground: .white, background: .bcsGreen)
        case "B":
            return GradeColorInfo(foreground: .white, background: .bcsGold)
        case "C":
            return GradeColorInfo(foreground: .white, background: .bcsOrange)
        default:
            return GradeColorInfo(foreground: .white, background: .bcsRed)
        }
    }
}

struct GradeColorInfo {
    let foreground: SwiftUI.Color
    let background: SwiftUI.Color
}

import SwiftUI

File: android/BestChanceStudio/app/src/main/kotlin/org/bestchancestudio/app/utils/GradeCalculator.kt
package org.bestchancestudio.app.utils

import androidx.compose.ui.graphics.Color
import org.bestchancestudio.app.models.GradeThreshold
import org.bestchancestudio.app.ui.theme.BcsGold
import org.bestchancestudio.app.ui.theme.BcsGreen
import org.bestchancestudio.app.ui.theme.BcsOrange
import org.bestchancestudio.app.ui.theme.BcsRed

data class GradeResult(val grade: String, val label: String)

data class GradeColorInfo(val foreground: Color, val background: Color)

object GradeCalculator {
    val gradeOrder = listOf("A+", "A", "B", "C", "D")

    fun grade(totalScore: Int, thresholds: Map<String, GradeThreshold>): GradeResult {
        for (g in gradeOrder) {
            val t = thresholds[g] ?: continue
            if (totalScore >= t.min && totalScore <= t.max) {
                return GradeResult(g, t.label)
            }
        }
        return GradeResult("D", thresholds["D"]?.label ?: "Needs significant improvement.")
    }

    fun gradeColor(grade: String): GradeColorInfo = when (grade) {
        "A+", "A" -> GradeColorInfo(Color.White, BcsGreen)
        "B" -> GradeColorInfo(Color.White, BcsGold)
        "C" -> GradeColorInfo(Color.White, BcsOrange)
        else -> GradeColorInfo(Color.White, BcsRed)
    }
}

fun scoreColor(score: Int): Color = when (score) {
    2 -> BcsGreen
    1 -> BcsGold
    else -> BcsRed
}
